### DISCUSSION TITLE
How to getList by array of ids?

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2463] [Q&A] [answered]
Is it possible to getList array of ids in frontend? or Should I write http function that query by array of ids in backend?

**Comment by ganigeorgiev** [comment, marked as answer]
You can use `||` to combine multiple id conditions, something like:

```js
const ids = ["id1", "id2", "id3"];

const result = await pb.collection("example").getList(1, 20, {
    // aka. `id="id1" || id="id2" || id="id3"`
    filter: ids.map((id) => `id="${id}"`).join("||"),
})
```

**Reply by 0q2** [reply]
I guess this works, but there really should be a builtin for this.

**Reply by gagefox** [reply]
If I pass this an empty array I get back everything in the collection. Is there a way to make it so I receive nothing?

**Reply by ganigeorgiev** [reply]
@gagefox You could skip sending the request or if for some reason you can't - send a dummy filter that will always evaluate to false, like `1=2`.
If you want the user to be required to provide an `id` filter, another possible solution could be to require it on List API rule level with something like `@request.query.filter ~ "id="`.
